CRM/Logging/Schema.php

```diff
@@ -302,23 +310,36 @@ public function fixSchemaDifferences($enableLogging = FALSE) {
    * and also implements the engine change defined by the hook (i.e. INNODB).
    *
    * Note changing engine & adding hook-defined indexes, but not changing back
-   * to ARCHIVE if engine has not been deliberately set (by hook) and not dropping
-   * indexes. Sysadmin will need to manually intervene to revert to defaults.
+   * to INNODB if engine has not been deliberately set (by hook) and not
+   * dropping indexes. Sysadmin will need to manually intervene to revert to
+   * defaults.
    *
    * @param array $params
-   *     'updateChangedEngineConfig' - update if the engine config changes, default FALSE
+   *     'updateChangedEngineConfig' - update if the engine config changes,
+   *                                   default FALSE
+   *     'forceEngineMigration' - force engine upgrade from ARCHIVE to InnoDB,
+   *                              default FALSE
    *
    * @return int $updateTablesCount
+   * @throws \CiviCRM_API3_Exception
    */
   public function updateLogTableSchema($params = []) {
     isset($params['updateChangedEngineConfig']) ? NULL : $params['updateChangedEngineConfig'] = FALSE;
+    isset($params['forceEngineMigration']) ? NULL : $params['forceEngineMigration'] = FALSE;
```

Comment:
Stylistically I have a problem with this - 

```
$params['forceEngineMigration'] =  isset($params['forceEngineMigration']) ? $params['forceEngineMigration'] : FALSE ;

```

would be more consistent with our style elsewhere (I realise the style could be argued but that's consistent with our general code style). However there are 2 other options

1) null coalescing operator now we are on 7.x....

```
$params['formEngineMigration'] = $params['formEngineMigration'] ?? FALSE;
```

I would also consider setting as an api.default value